dts/bindings/pinctrl/ambiq,apollo4-pinctrl.yaml

```diff
@@ -108,4 +108,70 @@ child-binding:
         type: int
         default: 0
         description: |
-          IOM nCE module select. The default value is reset value.
+          IOM nCE module select, selects the SPI channel (CE) number (0-42).
+            0,     IOM0CE0 : IOM 0 NCE 0 module
```

Comment:
@gmarull would it be better if these were strings? I don't want to give myself any more work but it would have less wording.

Reply:
It's ok as is, you may provide defines to ease reading DT